### PR TITLE
Fix x509 error about vendor link

### PR DESCRIPTION
**Pull request by windsonsea**
Hi @chalin , as you suggested in #2128, the `https` certificate was updated months ago.  

Now change the url to `Insight` that is an observability product and integrates with many OTel features. 

image

### CODE REVIEW

File: content/en/ecosystem/vendors.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ OpenTelemetry in their commercial products.
 | AWS                        | Yes               | No          | [aws-otel.github.io](https://aws-otel.github.io)                                                                                                      |
 | Azure                      | Yes               | No          | [docs.microsoft.com/...](https://docs.microsoft.com/azure/azure-monitor/app/opentelemetry-overview)                                                   |
 | Coralogix                  | Yes               | Yes         | [coralogix.com/...](https://coralogix.com/docs/opentelemetry/)                                                                                        |
-| DaoCloud                   | Yes               | Yes         | [docs.daocloud.io/...](https://docs.daocloud.io/en/insight/06UserGuide/01quickstart/otel/otel/)                                                       |
+| DaoCloud                   | Yes               | Yes         | [docs.daocloud.io/...](https://docs.daocloud.io/en/insight/03ProductBrief/WhatisInsight/)                                                       |
```
Comment (chalin) [inline content/en/ecosystem/vendors.md:21]: As @cartermp mentioned elsewhere, we prefer links into the OTel docs, so the original link was better.

**Review by chalin — CHANGES_REQUESTED** [review]
I'm still see an invalid certificate (notice the expiry date):

> image

**Comment by windsonsea** [comment]
ok, well noted. I'm asking IT staff why. This PR can be closed because nothing need to be changed currently.

The certificate looks different in and out of China -_! 

image

**Comment by chalin** [comment]
> The certificate looks different in and out of China -_!

That is what I was beginning to suspect.

**Comment by windsonsea** [comment]
Finally, the SSL certificate works now. It is effective after 24:00 last night while it was updated 8 hours ago.

image

Verified it at: https://www.sslchecker.com/sslchecker

**Comment by chalin** [comment]
Thanks for the confirmation. I've removed the link-checker ignore rule for daocloud.io in #2362, see https://github.com/open-telemetry/opentelemetry.io/pull/2362/files#diff-9ad64044531d7d8c033064f11141586478b535147312374c24f8d2d997722889L19-L20.

**Pull request closed without merging**